declarations/WebpackOptions.d.ts

```diff
@@ -0,0 +1,1249 @@
+				/**
+				 * add information about modules nested in other modules (like with module concatenation)
+				 */
+				nestedModules?: boolean;
+				/**
+				 * add information about assets inside modules
+				 */
+				moduleAssets?: boolean;
+				/**
+				 * add children information
+				 */
+				children?: boolean;
+				/**
+				 * add also information about cached (not built) modules
+				 */
+				cached?: boolean;
+				/**
+				 * Show cached assets (setting this to `false` only shows emitted files)
+				 */
+				cachedAssets?: boolean;
+				/**
+				 * add information about the reasons why modules are included
+				 */
+				reasons?: boolean;
+				/**
+				 * add the source code of modules
+				 */
+				source?: boolean;
+				/**
+				 * add warnings
+				 */
+				warnings?: boolean;
+				/**
+				 * add errors
+				 */
+				errors?: boolean;
+				/**
+				 * Suppress warnings that match the specified filters. Filters can be Strings, RegExps or Functions
+				 */
+				warningsFilter?: FilterTypes;
+				/**
+				 * Suppress assets that match the specified filters. Filters can be Strings, RegExps or Functions
+				 */
+				excludeAssets?: FilterTypes;
+				/**
+				 * Suppress modules that match the specified filters. Filters can be Strings, RegExps, Booleans or Functions
+				 */
+				excludeModules?: FilterTypes | boolean;
+				/**
+				 * Please use excludeModules instead.
+				 */
+				exclude?: FilterTypes | boolean;
+				/**
+				 * Display the entry points with the corresponding bundles
+				 */
+				entrypoints?: boolean;
+				/**
+				 * Display all chunk groups with the corresponding bundles
+				 */
+				chunkGroups?: boolean;
+				/**
+				 * add details to errors (like resolving log)
+				 */
+				errorDetails?: boolean;
+				/**
+				 * add the origins of chunks and chunk merging info
+				 */
+				chunkOrigins?: boolean;
+				/**
+				 * sort the modules by that field
+				 */
+				modulesSort?: string;
+				/**
+				 * add dependencies and origin of warnings/errors
+				 */
+				moduleTrace?: boolean;
+				/**
+				 * sort the chunks by that field
+				 */
+				chunksSort?: string;
+				/**
+				 * sort the assets by that field
+				 */
+				assetsSort?: string;
+				/**
+				 * Add public path information
+				 */
+				publicPath?: boolean;
+				/**
+				 * Add output path information
+				 */
+				outputPath?: boolean;
+				/**
+				 * show exports provided by modules
+				 */
+				providedExports?: boolean;
+				/**
+				 * show exports used by modules
+				 */
+				usedExports?: boolean;
+				/**
+				 * show reasons why optimization bailed out for modules
+				 */
+				optimizationBailout?: boolean;
+		  }
+		| boolean
+		| ("none" | "errors-only" | "minimal" | "normal" | "detailed" | "verbose");
+	/**
+	 * Environment to build for
+	 */
+	target?:
+		| (
+				| "web"
+				| "webworker"
+				| "node"
+				| "async-node"
+				| "node-webkit"
+				| "electron-main"
+				| "electron-renderer")
+		| {
+				[k: string]: any;
```

Comment:
This is invalid and should be corresponding to: 

Original property:

Reply:
cc @bcherny, this looks like this might be a bug, I'll submit and reference this comment.

Reply:
Tied now to bcherny/json-schema-to-typescript#164

Reply:
I created https://github.com/bcherny/json-schema-to-typescript/pull/168 to address this.